hooker/api.py

```diff
@@ -40,15 +50,45 @@ def load(path):
     The function tries to import `example.module` when either `example.module`,
     `example/module` or `example/module.py` is given.
     """
+    if path.startswith("http://") or path.startswith("https://"):
+        import httpimport
+        url, module = path.split("/", -1)
+        with httpimport.remote_repo(url):
+            __import__(module)
+        return
+
+        raise Exception("Loading plugins through HTTP/S is not implemented.")
+
+
+    logger.info("Loading plugin at '%s'" % (path))
 
     importpath = path.replace("/", ".").replace("\\", ".")
-    if importpath[-3:] == ".py":
-        importpath = importpath[:-3]
+    if importpath.endswith(".py"):  # Remove the file extension
+        importpath = ".".join(importpath.split(".")[:-1])
+        logger.info("Plugin module at '%s'" % (importpath))
 
     try:
-        importlib.import_module(importpath)
+        if overwrite and importpath in sys.modules:
+            logger.info("Overwriting plugin '%s'!" % (importpath))
+            importlib.reload(importpath)
+        else:
+            importlib.import_module(importpath)
     except (ModuleNotFoundError, ImportError, TypeError):
+        logger.warning("Failed loading '%s' as module ('%s'). Trying execution as file..." % (path, importpath))
+
         try:
-            exec(open(path).read())
+            with open(path) as f:
+                exec(f.read())
         except FileNotFoundError:
-            logger.error("Module or file %s was not found! Ignoring...", importpath)
+            logger.error("Module or file '%s' was not found! Ignoring...", importpath)
```

Comment:
This should raise an exception, `logger.error` will not halt execution (while it probably should?)